lib/src/debug.dart

```diff
@@ -127,30 +126,29 @@ void dumpChunks(int start, List<Chunk> chunks) {
       }
     }
 
-    if (chunk.rule == null) {
+    var rule = chunk.rule;
+    if (rule == null) {
       row.add('');
       row.add('(no rule)');
       row.add('');
     } else {
-      writeIf(chunk.rule.cost != 0, () => '\$${chunk.rule.cost}');
+      writeIf(rule.cost != 0, () => '\$${rule.cost}');
 
-      var ruleString = chunk.rule.toString();
-      if (chunk.rule.isHardened) ruleString += '!';
+      var ruleString = rule.toString();
+      if (rule.isHardened) ruleString += '!';
       row.add(ruleString);
 
-      var constrainedRules =
-          chunk.rule.constrainedRules.toSet().intersection(rules);
+      var constrainedRules = rule.constrainedRules.toSet().intersection(rules);
       writeIf(constrainedRules.isNotEmpty,
           () => "-> ${constrainedRules.join(" ")}");
     }
 
     writeIf(chunk.indent != null && chunk.indent != 0,
         () => 'indent ${chunk.indent}');
 
-    writeIf(chunk.nesting != null && chunk.nesting.indent != 0,
-        () => 'nest ${chunk.nesting}');
+    writeIf(chunk.nesting?.indent != 0, () => 'nest ${chunk.nesting}');
```

Comment:
If `chunck.nesting` is `null`, then this will write `nest null`.  I don't think you want that behavior.  I think you want:

```dart
writeIf(chunk.nesting != null && chunk.nesting!.indent != 0, ...);
```

or maybe this is better? (I can't decide if it's too clever):

```dart
writeIf((chunk.nesting?.indent ?? 0) != 0, ...);
```

Reply:
Good catch. I had tried to make `nesting` non-nullable and thus deleted the null check here. But then I ended up backing out that change. But I just tried again and was able to make it non-nullable so the null check here is indeed no longer needed.